machineconfiguration/v1alpha1/types_machineconfignode.go

```diff
@@ -0,0 +1,151 @@
+	// +kubebuilder:validation:Required
+	Spec MachineConfigNodeSpec `json:"spec"`
+
+	// status describes the last observed state of this machineconfignode
+	// +optional
+	Status MachineConfigNodeStatus `json:"status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// MachineConfigNodeList describes all of the MachinesStates on the system
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type MachineConfigNodeList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []MachineConfigNode `json:"items"`
+}
+
+// MCOObjectReference holds information about an object the MCO either owns
+// or modifies in some way
+type MCOObjectReference struct {
+	// name describes what this object is called
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+}
+
+// MachineConfigNodeSpec describes the type of State we are managing
+type MachineConfigNodeSpec struct {
+	// node contains references to the node for this MachineConfigNode
+	// +kubebuilder:validation:Required
+	Node MCOObjectReference `json:"node"`
+
+	// pool contains a reference to the MachineConfigPool that this MachineConfigNode's referenced Node belongs to.
+	// +kubebuilder:validation:Required
+	Pool MCOObjectReference `json:"pool"`
+
+	// configVersion holds the desired config version for the node targeted by this MachineConfigNode resource.
+	// The desired version represents the machine config the node will attempt to update to when it next has the chance.
+	// +kubebuilder:validation:Required
+	ConfigVersion MCNSpecMachineConfigVersion `json:"configVersion"`
+}
+
+// MachineConfigNodeStatus holds the reported information on a particular MachineConfigNode
+type MachineConfigNodeStatus struct {
+	// conditions represent the observations of a MachineConfigNode's current state.
+	// Known .status.conditions.type are: "UpdatePreparing", "UpdateInProgress", "UpdatePostAction", "UpdateCompleting", "Resuming", "Updated",
+	// "Ready", "ComparingMCs", "DrainingNode", "ApplyingFilesAndOS", "CordoningNode", "RebootingNode", and "ReloadingCRIO"
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+	// observedGeneration represents the generation observed by the controller.
+	// +kubebuilder:validation:Required
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+	// configVersion describes the current and desired MachineConfig for this node.
+	// The current version represents the current machine config for the node and is updated after a successful update.
+	// The desired version represents the machine config the node will attempt to update to when it next has the chance.
```

Comment:
What does has the chance mean? Does it not happen as soon as the values differ?

Reply:
sorry, this was bad wording. Will remove.